pkg/quota/v1/evaluator/core/services.go

```diff
@@ -120,22 +120,36 @@ func (p *serviceEvaluator) Usage(item runtime.Object) (corev1.ResourceList, erro
 	// default service usage
 	result[serviceObjectCountName] = *(resource.NewQuantity(1, resource.DecimalSI))
 	result[corev1.ResourceServices] = *(resource.NewQuantity(1, resource.DecimalSI))
-	result[corev1.ResourceServicesLoadBalancers] = resource.Quantity{Format: resource.DecimalSI}
-	result[corev1.ResourceServicesNodePorts] = resource.Quantity{Format: resource.DecimalSI}
+	// check nodeport and load balancer only if the runtime svc is one of them.
 	switch svc.Spec.Type {
 	case corev1.ServiceTypeNodePort:
 		// node port services need to count node ports
 		value := resource.NewQuantity(int64(ports), resource.DecimalSI)
 		result[corev1.ResourceServicesNodePorts] = *value
 	case corev1.ServiceTypeLoadBalancer:
-		// load balancer services need to count node ports and load balancers
-		value := resource.NewQuantity(int64(ports), resource.DecimalSI)
-		result[corev1.ResourceServicesNodePorts] = *value
+		// load balancer services need to count node ports. If creation of node ports
+		// is suspressed only ports with explicit NodePort values are counted.
+		if svc.Spec.AllocateLoadBalancerNodePorts == nil || *svc.Spec.AllocateLoadBalancerNodePorts == true {
```

Comment:
should we check for the featuregate?
` if utilfeature.DefaultFeatureGate.Enabled(features.ServiceLBNodePortControl) {`

Reply:
@aojea ,  if the feature gate is enabled, the `AllocateLoadBalancerNodePorts` must be not nil
If not enabled, we can check whether it is nil before checking it's true or false.

I think both are OK to me. 

However, checking the feature gate would be friendly to code reading. So I would like to check feature gate as you suggested.


```
		if feature.DefaultFeatureGate.Enabled(features.ServiceLBNodePortControl) &&
			*svc.Spec.AllocateLoadBalancerNodePorts == false {
			result[corev1.ResourceServicesNodePorts] = *portsWithNodePorts(svc)
		} else {
			value := resource.NewQuantity(int64(ports), resource.DecimalSI)
			result[corev1.ResourceServicesNodePorts] = *value
		}
```

or

```
		if svc.Spec.AllocateLoadBalancerNodePorts == nil || *svc.Spec.AllocateLoadBalancerNodePorts == true {
			value := resource.NewQuantity(int64(ports), resource.DecimalSI)
			result[corev1.ResourceServicesNodePorts] = *value
		} else {
			result[corev1.ResourceServicesNodePorts] = *portsWithNodePorts(svc)
		}
```

Reply:
the question is for someone like @liggitt ^^^